hw_7/DOM.js

```diff
@@ -0,0 +1,301 @@
+        event.target.closest('li').remove();
+    }
+    setItem('data_myUl', ul_td.innerHTML);
+}
+
+function onDelAllLi(event) {
+    if (event.target.className === 'deleteAll') {
+        ul_td.innerHTML = '';
+    }
+    setItem('data_myUl', ul_td.innerHTML);
+}
+
+function onChecked(event) {
+    if (event.target.className === 'check') {
+        event.target.checked ?
+            event.target.nextElementSibling.style.textDecoration = 'line-through' :
+            event.target.nextElementSibling.style.textDecoration = 'none';
+
+    }
+}
+
+function onStor(event) {
+    if (event.key === 'data_myUl') {
+        ul_td.innerHTML = `${event.newValue}`
+    }
+}
+
+function setItem(key, item) {
+    try {
+        localStorage.setItem(key, item);
+    }
+    catch (e) {
+        console.error(e)
+    }
+}
+
+function getItem(key) {
+    try {
+        return localStorage.getItem(key);
+    }
+    catch (e) {
+        console.error(e)
+    }
+}
+
+document.querySelector('.add').addEventListener('click', onAddLi);
+document.querySelector('.deleteAll').addEventListener('click', onDelAllLi);
+document.querySelector('ul').addEventListener('click', onDelLi);
+//document.querySelector('ul').addEventListener('change', onChecked); //onCheck
+window.addEventListener('storage', onStor);
+*/
+
+
+// Вариант № 2
+//ul_td.innerHTML = getItem('data_myUl');
+
+let myLi = [];
+let htmlStr = '';
+let maxId = getMax(myLi);
+
+
+if (getItem('data_myUl')){
+   let myLi = getItem('data_myUl');
+   render(myLi);
+}
+
+function getMax(myObj) {
+    let tempPer = 0;
+    for (let i = 0; i < myObj.length; i++) {
+        tempPer = Math.max(tempPer, myObj[i].id);
+    }
+    return tempPer;
+}
+
+
+function addLi(text) {
+    let myTemp = {
+        id: ++maxId,
+        text: text,
+        data: new Date().toLocaleString(),
+        isCheck: false,
+    }
+
+    myLi.push(myTemp);
+
+    htmlStr += addLihtml(maxId, text, myTemp.data, myTemp.isCheck);
+}
+function addLihtml(id, text, data, isCheck) {
+    return `<li id="${id}">
+                        <div class="todo-li">
+                            <div>
+                                <input type="checkbox" class="check" ${isCheck ? 'checked' : ''}>
+                                <div><span class="doToText">${text}</span></div>
+                                <button class="close">x</button>
+                            </div>
+                            <div>
+                                <div><span class="date">${data}</span></div>
+                            </div>
+                        </div>
+                    </li>`
+}
+
+function delLi(id) {
+    myLi = myLi.filter(item => item.id != id);
+    render(myLi);
+}
+
+function render(myLi) {
+    htmlStr = '';
+    if (myLi.length) {
+        //return ul_td.innerHTML = '';
+        for (let i = 0; i < myLi.length; i++) {
+        htmlStr += addLihtml(myLi[i].id, myLi[i].text, myLi[i].data, myLi[i].isCheck);
+         }
```

Comment:
может лучше сделать через for of?
так читабельней будет
```suggestion
        for (const li of myLi) {
            htmlStr += addLihtml(li.id, li.text, li.data, li.isCheck);
        }
```